app/src/main/java/com/kms/cura/view/fragment/PatientHomeFragment.java

```diff
@@ -0,0 +1,211 @@
+
+
+    public PatientHomeFragment() {
+        // Required empty public constructor
+    }
+
+    public static PatientHomeFragment newInstance(Context mContext, Activity activity) {
+        PatientHomeFragment fragment = new PatientHomeFragment();
+        fragment.setContext(mContext);
+        fragment.setActivity(activity);
+        return fragment;
+    }
+
+    private void setContext(Context src) {
+        this.mContext = src;
+    }
+
+    private void setActivity(Activity src) {
+        this.activity = src;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View root = inflater.inflate(R.layout.fragment_patient_home, container, false);
+        updateSpinner = this;
+        initView(root);
+        setUpSpnSpeciality(root);
+        return root;
+    }
+
+    public void initView(View root) {
+        edtName = (EditText) root.findViewById(R.id.edtName);
+        edtLocation = (EditText) root.findViewById(R.id.edtLocation);
+        rdbtngroupLocation = (RadioGroup) root.findViewById(R.id.rdbtngroupLoacation);
+        rdbtngroupLocation.setOnCheckedChangeListener(PatientHomeFragment.this);
+        rdbtnCurrentLocation = initRadioButton(root, R.id.rdbtnCurrentLocation, 1);
+        rdbtnManualEnter = initRadioButton(root, R.id.rdbtnManuallyEnter, 2);
+    }
+
+    public RadioButton initRadioButton(View root, int id, int newId) {
+        RadioButton tmp = (RadioButton) root.findViewById(id);
+        tmp.setId(newId);
+        return tmp;
+    }
+
+    public void initButton(View root){
+        btnRegister = (Button) root.findViewById(R.id.btnRegister);
+        btnRegister.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if(!checked){
+                    currentLocation = edtLocation.getText().toString();
+                }
+                //Search Function implement here
+            }
+        });
+    }
+
+    public void setUpSpnSpeciality(View root){
+        spnSpeciality = (Spinner) root.findViewById(R.id.spnSpeciality);
+        //Place holder for speciality
+        speciality = new ArrayList<>();
+        speciality.add("A");
+        speciality.add("B");
+        speciality.add("C");
+        speciality.add("Please choose");
+        //------------------
+        specialityAdapter = new CheckBoxAdapter(getActivity(),R.layout.check_box_item,speciality,specialitySelected,updateSpinner);
+        spnSpeciality.setAdapter(specialityAdapter);
+        spnSpeciality.setSelection(specialityAdapter.getCount());
+    }
+
+    @Override
+    public void onCheckedChanged(RadioGroup group, int checkedId) {
+        switch (checkedId) {
+            case 1:
```

Comment:
Extract 1 and 2 below to a constant with a meaningful name to help understand what it's mean